Small warning cleanup

icco opened the pull request:
- Let people specify `name` and `name_prefix` in options
- Change the `name` function to not fire warnings that `name` was not initialized.
- Preinitialize `@exit` to nil so it doesn't warn as well.

This is to fix the following warnings:

```
~/.rvm/gems/ruby-2.3.0/bundler/gems/delayed_job-20d40edf12a8/lib/delayed/worker.rb:148: warning: instance variable @name not initialized
~/.rvm/gems/ruby-2.3.0/bundler/gems/delayed_job-20d40edf12a8/lib/delayed/worker.rb:151: warning: instance variable @name_prefix not initialized
~/.rvm/gems/ruby-2.3.0/bundler/gems/delayed_job-20d40edf12a8/lib/delayed/worker.rb:148: warning: instance variable @name not initialized
~/.rvm/gems/ruby-2.3.0/bundler/gems/delayed_job-20d40edf12a8/lib/delayed/worker.rb:151: warning: instance variable @name_prefix not initialized
~/.rvm/gems/ruby-2.3.0/bundler/gems/delayed_job-20d40edf12a8/lib/delayed/worker.rb:148: warning: instance variable @name not initialized
~/.rvm/gems/ruby-2.3.0/bundler/gems/delayed_job-20d40edf12a8/lib/delayed/worker.rb:151: warning: instance variable @name_prefix not initialized
~/.rvm/gems/ruby-2.3.0/bundler/gems/delayed_job-20d40edf12a8/lib/delayed/worker.rb:148: warning: instance variable @name not initialized
~/.rvm/gems/ruby-2.3.0/bundler/gems/delayed_job-20d40edf12a8/lib/delayed/worker.rb:151: warning: instance variable @name_prefix not initialized
~/.rvm/gems/ruby-2.3.0/bundler/gems/delayed_job-20d40edf12a8/lib/delayed/worker.rb:206: warning: instance variable @exit not initialized
```

albus522 commented:
Your implementation completely changes how `name` operates. The before and after is not equivalent.

Also this isn't really a warning we care about, nor do most people.

Lastly what are you trying to accomplish with the name/prefix in the options hash?

icco commented:
@albus522 How are they different? The only difference that I can tell of is the name would be stored in `@name` after the first call to `#name`. Is this an issue? If so, I could write this using the less commonly used `defined?`.

And warnings in my build log bother me a lot.

icco commented:
Oh and the options hash change just seemed like an oversight. Seemed like something someone would want when initializing a worker.

albus522 commented:
Yes, not memoizing is intentional.

Rails alone spits out dozens of warning. You are in for a very up hill battle.

Generally speaking, users don't initialize the worker. DJ does in the various runner commands. So if we aren't using it, there likely isn't a need.

icco commented:
I'm not using DJ with Rails. But if you prefer the warnings there, I guess sorry for taking the time.

I've refactored to not store anything (besides nil) in `@name` if not explicitly defined. Also removed the options hash stuff.
